extension/encoding/awslogsencodingextension/internal/unmarshaler/cloudtraillogs/cloudtraillogs.go

```diff
@@ -0,0 +1,290 @@
+	EventTime                    string         `json:"eventTime"`
+	EventSource                  string         `json:"eventSource"`
+	EventName                    string         `json:"eventName"`
+	AwsRegion                    string         `json:"awsRegion"`
+	SourceIPAddress              string         `json:"sourceIPAddress"`
+	UserAgent                    string         `json:"userAgent"`
+	RequestID                    string         `json:"requestID"`
+	EventID                      string         `json:"eventID"`
+	EventType                    string         `json:"eventType"`
+	EventCategory                string         `json:"eventCategory"`
+	RecipientAccountID           string         `json:"recipientAccountId"`
+	UserIdentity                 map[string]any `json:"userIdentity"`
+	ResponseElements             map[string]any `json:"responseElements"`
+	RequestParameters            map[string]any `json:"requestParameters"`
+	Resources                    []any          `json:"resources"`
+	ReadOnly                     bool           `json:"readOnly"`
+	ManagementEvent              bool           `json:"managementEvent"`
+	AdditionalEventData          map[string]any `json:"additionalEventData"`
+	TLSDetails                   map[string]any `json:"tlsDetails"`
+	SessionCredentialFromConsole string         `json:"sessionCredentialFromConsole"`
+}
+
+type CloudTrailLogs struct {
+	Records []CloudTrailRecord `json:"Records"`
+}
+
+func NewCloudTrailLogsUnmarshaler(buildInfo component.BuildInfo) plog.Unmarshaler {
+	return &CloudTrailLogsUnmarshaler{
+		buildInfo: buildInfo,
+		gzipPool:  sync.Pool{},
+	}
+}
+
+func (u *CloudTrailLogsUnmarshaler) Unmarshal(buf []byte) (plog.Logs, error) {
+	return u.UnmarshalLogs(buf)
+}
+
+func (u *CloudTrailLogsUnmarshaler) UnmarshalLogs(buf []byte) (plog.Logs, error) {
+	gzipReader, ok := u.gzipPool.Get().(*gzip.Reader)
+	if !ok {
+		var err error
+		gzipReader, err = gzip.NewReader(bytes.NewReader(buf))
+		if err != nil {
+			return plog.Logs{}, fmt.Errorf("failed to create gzip reader: %w", err)
+		}
+	} else {
+		if err := gzipReader.Reset(bytes.NewReader(buf)); err != nil {
+			u.gzipPool.Put(gzipReader)
+			return plog.Logs{}, fmt.Errorf("failed to reset gzip reader: %w", err)
+		}
+	}
+
+	defer func() {
+		_ = gzipReader.Close()
+		u.gzipPool.Put(gzipReader)
+	}()
+
+	decompressedBuf, err := io.ReadAll(gzipReader)
+	if err != nil {
+		return plog.Logs{}, fmt.Errorf("failed to decompress CloudTrail logs: %w", err)
+	}
+
+	var cloudTrailLogs CloudTrailLogs
+	if err := json.Unmarshal(decompressedBuf, &cloudTrailLogs); err != nil || len(cloudTrailLogs.Records) == 0 {
+		return plog.Logs{}, fmt.Errorf("failed to unmarshal CloudTrail logs: %w", err)
+	}
+
+	return u.processRecords(cloudTrailLogs.Records), nil
+}
+
+func (u *CloudTrailLogsUnmarshaler) processRecords(records []CloudTrailRecord) plog.Logs {
+	logs := plog.NewLogs()
+
+	if len(records) == 0 {
+		return logs
+	}
```

Comment:
I think that it makes sense to have the validation here instead of the unmarshaling one. WDYT? Relates to my previous comment